aws/sdk/integration-tests/s3/tests/streaming-response.rs

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::{Credentials, Endpoint, Region};
+use bytes::BytesMut;
+use std::future::Future;
+use std::net::SocketAddr;
+use std::time::Duration;
+use tracing::debug;
+
+// test will hang forever with the default (single-threaded) test executor
+#[tokio::test(flavor = "multi_thread")]
+#[should_panic(
+    expected = "error reading a body from connection: end of file before message length reached"
```

Comment:
Only thing I'd note is that it can be fragile to make the unit tests depend on the string output of an error (it's not stable, we could reword to improve understanding).